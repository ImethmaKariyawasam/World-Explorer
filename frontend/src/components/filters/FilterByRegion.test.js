import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import FilterByRegion from './FilterByRegion';

describe('FilterByRegion Component', () => {
  const mockOnFilter = jest.fn();

  beforeEach(() => {
    mockOnFilter.mockClear();
  });

  it('renders with all region options', () => {
    render(<FilterByRegion onFilter={mockOnFilter} />);
    
    expect(screen.getByRole('combobox')).toBeInTheDocument();
    expect(screen.getByText('All')).toBeInTheDocument();
    expect(screen.getByText('Africa')).toBeInTheDocument();
    expect(screen.getByText('Americas')).toBeInTheDocument();
    expect(screen.getByText('Asia')).toBeInTheDocument();
    expect(screen.getByText('Europe')).toBeInTheDocument();
    expect(screen.getByText('Oceania')).toBeInTheDocument();
  });

  it('calls onFilter when region is selected', () => {
    render(<FilterByRegion onFilter={mockOnFilter} />);
    
    fireEvent.change(screen.getByRole('combobox'), { target: { value: 'asia' } });
    expect(mockOnFilter).toHaveBeenCalledWith('asia');
  });

  it('defaults to All option', () => {
    render(<FilterByRegion onFilter={mockOnFilter} />);
    
    expect(screen.getByDisplayValue('All')).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<FilterByRegion onFilter={mockOnFilter} />);
    expect(asFragment()).toMatchSnapshot();
  });
});