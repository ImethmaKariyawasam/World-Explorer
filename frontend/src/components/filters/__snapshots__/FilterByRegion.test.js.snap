// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterByRegion Component matches snapshot 1`] = `
<DocumentFragment>
  <select
    aria-label="Filter by region"
    class="form-select"
    style="width: 200px;"
  >
    <option
      value=""
    >
      All
    </option>
    <option
      value="africa"
    >
      Africa
    </option>
    <option
      value="americas"
    >
      Americas
    </option>
    <option
      value="asia"
    >
      Asia
    </option>
    <option
      value="europe"
    >
      Europe
    </option>
    <option
      value="oceania"
    >
      Oceania
    </option>
  </select>
</DocumentFragment>
`;
